client_authentication_test.go

```diff
@@ -509,6 +510,31 @@ func TestAuthenticateClient(t *testing.T) {
 			r:         new(http.Request),
 			expectErr: ErrInvalidClient,
 		},
+		{
+			d:      "should pass because assertion contains client id as issuer and client is public",
```

Comment:
Can you please also add some failing tests?

Reply:
I have added the following tests:

- Pass: should pass because assertion contains client id as issuer and client is public (Testing the added functionality)
- Fail: should fail despite valid assertion because client is private (testing whether behaviour is consistent with config)
- Fail: should fail despite valid assertion because client unknown (testing whether behaviour is consistent with config)